### PR TITLE
Removed the Files API from the Gallery

### PR DESCRIPTION
**Pull request by William-Byrne**
I upgraded the gallery back end from the Files API to the Google Cloud Store. An appspot instance with my changes can be found here: [http://galleryioupgrade.appspot.com/](http://galleryioupgrade.appspot.com/).

@halatmit @jisqyv @afmckinney

### CODE REVIEW

File: appinventor/appengine/src/com/google/appinventor/server/ProjectServiceImpl.java [hunk 1]
```diff
@@ -568,21 +569,20 @@ public long addFile(long projectId, String fileId) {
...
-    boolean lockForRead = false;
+  public UserProject newProjectFromGallery(String projectName, String galleryPath,
+      long galleryId) {
     try {
-      FileService fileService = FileServiceFactory.getFileService();
-      AppEngineFile readableFile = new AppEngineFile(aiaPath);
-      FileReadChannel readChannel = fileService.openReadChannel(readableFile, false);
+      GcsService fileService = GcsServiceFactory.createGcsService();
+      GcsFilename readableFile = new GcsFilename(Flag.createFlag("gallery.bucket", "").get(), galleryPath);
+      GcsInputChannel readChannel = fileService.openPrefetchingReadChannel(readableFile, 0, 8000);
```
Comment (jisqyv) [inline appinventor/appengine/src/com/google/appinventor/server/ProjectServiceImpl.java]: I would recommend a buffer size which is  a power of 2, say 8192 or 16384.

**Comment by jisqyv** [comment]
@William-Byrne See the line note...

**Comment by William-Byrne** [comment]
Sure, I'll make it happen. 2^14 seems like a good choice.

**Comment by William-Byrne** [comment]
Good catch.

@jisqyv

**Comment by jisqyv** [comment]
@William-Byrne That would be great.

**Comment by William-Byrne** [comment]
@jisqyv All set.

**Comment by jisqyv** [comment]
LGTM (last change above made). Moving to Gerrit for Master

**Comment by jisqyv** [comment]
MERGED

**Pull request closed without merging**